src/crypto/Address.dto.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+
+import {
+  Address,
+  ByronAddress,
+  BaseAddress,
+  PointerAddress,
+  EnterpriseAddress,
+  RewardAddress,
+} from '@emurgo/react-native-haskell-shelley'
+import {Logger} from '../utils/logging'
+
+// NOTE: ** IMPORTANT ** The order matters
+const ADDRESS_TYPES = Object.freeze([
+  ByronAddress,
+  BaseAddress,
+  PointerAddress,
+  EnterpriseAddress,
+  RewardAddress,
+])
+
+export type KeyHashesCardano = {|
+  spending: string,
+  staking: string,
+|}
+
+/**
+ * @description The resolver for the Cardano key hashes
+ * @param {Address} wasmAddress A Cardano wallet Address
+ * @returns KeyHashesCardano|null You don't need to test for Byron it will return null
+ */
+export const getKeyHashesCardano = async (
+  wasmAddress: Address,
+): Promise<KeyHashesCardano | null> => {
+  let addr, pay, stake, paymentKeyHash, stakingKeyHash
+  for (const [i, addressType] of ADDRESS_TYPES.entries()) {
```

Comment:
Probably instead of relying on a for sequence that can be broken if the `ADDRESS_TYPES` object is modified, it might be better to use the `normalizeToAddress` function

Reply:
The object is frozen but moving to normalize that already covers that.